### PR TITLE
Fix for issue #840

### PR DESCRIPTION
**Pull request by iz-iznogood**
Add support for indices in EntitySchema and EntitySchemaTransformer

### CODE REVIEW

**Comment by iz-iznogood** [comment]
Please review and provide feedback

Thanks

File: src/entity-schema/EntitySchema.ts [hunk 1]
```diff
@@ -247,4 +247,22 @@ export interface EntitySchema { // todo: make it-to-date
...
+
+            /**
+             * Index column names.
+             */
+            columns: string[];
+
+            /**
+             * Index uniqueness.
+             */
+            unique: boolean;
```
Comment (pleerock) [inline src/entity-schema/EntitySchema.ts]: can you please add `sparse` option as well? (it is used in mongodb)

**Comment by pleerock** [comment]
Looks good, please add corresponding tests for this new functionality. We probably need separate directory for entity schema, create a new test in `test/functional/entity-schema/indices` directory

File: src/entity-schema/EntitySchema.ts (hunk 1 above)
Comment (ckalogeropoulos) [reply src/entity-schema/EntitySchema.ts]: Done

**Comment by iz-iznogood** [comment]
Hi, please pull

**Comment by pleerock** [comment]
Yes, and what about tests?

**Comment by iz-iznogood** [comment]
I added the test, hope it is ok.
If yes then please pull,

thanks

File: test/functional/entity-schema/indices/indices-test.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    describe("create index", function() {
+
+        it("should work without errors", () => Promise.all(connections.map(async connection => {
+
+            const queryRunner = connection.createQueryRunner();
+            const tableSchema = await queryRunner.loadTableSchema("person");
+            await queryRunner.release();
+
+            expect(tableSchema!.indices.length).to.be.equal(1);
+            expect(tableSchema!.indices[0].name).to.be.equal("IDX_UNQ_Person");
```
Comment (pleerock) [inline test/functional/entity-schema/indices/indices-test.ts]: what about to check what columns this index has?

**Comment by pleerock** [comment]
Looks good now :)

File: test/functional/entity-schema/indices/indices-test.ts (hunk 2 above)
Comment (iz-iznogood) [reply test/functional/entity-schema/indices/indices-test.ts]: DONE

**Comment by iz-iznogood** [comment]
Hi,

I need this merged so that I can send in more changes - fixes.
Please review and merge if possible.

Thanks

**Comment by iz-iznogood** [comment]
I will create a new pull request with fixes for #840, #833 and #834 since they are related
Sorry for this

**Pull request closed without merging**